src/selector.rs

```diff
@@ -0,0 +1,24 @@
+//! Abstract Faiss ID selector
+use faiss_sys::*;
+
+/// Abstraction over IDSelectorRange and IDSelectorBatch
+#[derive(Debug)]
+pub struct IdSelector {
```

Comment:
Don't forget to implement `Drop`, so that the inner selector object is freed in the end.

It should also be safe to `unsafe impl` `Send` and `Sync`, since it's a plain data structure with no additional concurrency concerns.